docs/client-sdk/javascript/tutorials/use-a-persistent-conversation-cache.md

```diff
@@ -0,0 +1,155 @@
+---
+sidebar_label: Use a persistent conversation cache
+sidebar_position: 7
+---
+
+# Use a persistent conversation cache to improve load time
+
+For a user with a large number of ongoing conversations, loading their conversation list in real-time can be a very expensive operation from the perspective of both bandwidth and CPU usage. This is because the message API client must download all of the [conversation invites](https://xmtp.org/docs/dev-concepts/invitation-and-message-encryption) and decrypt each one using X3DH before being able to display the conversations and their messages.
+
+To improve conversation load time, you can use the XMTP SDK to export the unencrypted conversation list in a JSON serializable format to a persistent conversation cache. For example, you can use the Zustand [`persist` middleware](https://github.com/pmndrs/zustand/blob/main/docs/integrations/persisting-store-data.md) to create a conversation cache that you can persist to the browser’s `LocalStorage`.
+
+:::info
+
+The process described in this tutorial will change soon. You can implement a persistent conversation cache as described here to address performance issues in your app now. However, please be prepared to update your app to use a different approach to be provided in the near future.
```

Comment:
should we provide rough timeline here, or unnecessary?

Reply:
@daria-github Thank you for such helpful feedback! I was a bit nervous about including this forward-looking statement here as we usually don't do that in tech docs. But I think in this case, we need to provide it as a courtesy to devs - and your feedback helped us make the info **truly** helpful. Updating this now to provide a timeline of Q2 2023.